carstore/read_write_blockstore.go

```diff
@@ -0,0 +1,66 @@
+package carstore
+
+import (
+	"sync"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2/blockstore"
+	"golang.org/x/xerrors"
+)
+
+// CarReadWriteStoreTracker tracks the lifecycle of a ReadWrite CAR Blockstore and makes it easy to create/get/cleanup the blockstores.
+// It's important to close a CAR Blockstore when done using it so that the backing CAR file can be closed.
+type CarReadWriteStoreTracker struct {
```

Comment:
Suggest refactoring to use one class for the implementation of CarReadWriteStoreTracker and CarReadOnlyStoreTracker if possible - the logic looks almost identical